runtime/list.c

```diff
@@ -2,7 +2,7 @@
 
 #include <string.h>
 
-#define ARRAY_LENGTH_MAX ((int64_t)(INT64_MAX/sizeof(TaggedPtr)))
+#define GENERIC_ARRAY_LENGTH_MAX ((int64_t)(INT64_MAX/sizeof(TaggedPtr)))
```

Comment:
Leave this as is. We are enforcing it for all arrays.

Reply:
I meant to increase the max size just for bytes up to INT64_MAX. Now I see I haven't done it correctly (should have changed `_bal_array_grow`), I'll revert this for now. 

But shouldn't we increase the max size for bytes?

Reply:
I see no point in increasing it. Just useless extra complexity.

We cannot go up to INT64_MAX anyway since we are using top 8 bits in pointers as a tag.